File: abc/135/a.py
A, B = map(int, input().split())

# A + B が偶数ならば A + B / 2, そうでなければ不可
ans = (
    (A + B) // 2 if (A + B) % 2 == 0 else
    'IMPOSSIBLE'
)

print(ans)

File: abc/135/a.rs
#![allow(non_snake_case)]
#![allow(unused_variables)]
#![allow(dead_code)]

fn main() {
    let (A, B): (i64, i64) = {
        let mut line: String = String::new();
        std::io::stdin().read_line(&mut line).unwrap();
        let mut iter = line.split_whitespace();
        (
            iter.next().unwrap().parse().unwrap(),
            iter.next().unwrap().parse().unwrap()
        )
    };

    let ans = if (A + B) % 2 == 0 { ((A + B) / 2).to_string() } else { "IMPOSSIBLE".to_string() };

    println!("{}", ans);
}
